### DISCUSSION TITLE
Why does using a string value for keys in Graphcache throws type error?

### DISCUSSION THREAD

**Discussion by GollyJer** [urql-graphql/urql#1224] [Q&A] [answered]
A few of our keys are strings.
This works but are we doing something wrong?
<br />

> <img width=800 src=https://user-images.githubusercontent.com/689204/101814675-509e6380-3ad3-11eb-84ce-37663412efec.png />

<br />

In [your types.ts file](https://github.com/FormidableLabs/urql/blob/4327263f26a27faa487c5ddee05ea77c5416a8db/exchanges/graphcache/src/types.ts) it looks like the type can be a Scalar or Primitive.

Why the warning?

**Comment by JoviDeCroock** [comment, marked as answer]
This relates to: https://github.com/FormidableLabs/urql/issues/901 we have no way to derive what type this is, you'll have to cast it to the type you want.
